packages/block-library/src/navigation-menu/block-colors-selector.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { IconButton, Dropdown, Toolbar } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { DOWN } from '@wordpress/keycodes';
+import { ColorPaletteControl, ContrastChecker } from '@wordpress/block-editor';
+
+/**
+ * Color Selector Icon component.
+ * @return {*} React Icon component.
+ * @constructor
+ */
+const ColorSelectorIcon = ( { style } ) =>
+	<div className="block-editor-block-colors-selector__icon-container">
+		<div
+			className="colors-selector__state-selection wp-block-navigation-menu-item"
+			style={ style }
+		>
+			{ __( 'Aa' ) }
+		</div>
+	</div>;
+
+/**
+ * Renders the Colors Selector Toolbar with the icon button.
+ *
+ * @param {function} onToggle Callback open/close Dropdown.
+ * @param {bool} isOpen True is the color settings dropdown is open. Otherwise, False.
+ * @return {*} React toggle button component.
+ */
+const renderToggle = ( style ) => ( { onToggle, isOpen } ) => {
+	const openOnArrowDown = ( event ) => {
+		if ( ! isOpen && event.keyCode === DOWN ) {
+			event.preventDefault();
+			event.stopPropagation();
+			onToggle();
+		}
+	};
+
+	return (
+		<Toolbar>
+			<IconButton
+				className="components-icon-button components-toolbar__control block-editor-block-colors-selector__toggle"
+				label={ __( 'Open Colors Selector' ) }
+				onClick={ onToggle }
+				onKeyDown={ openOnArrowDown }
+				icon={ <ColorSelectorIcon style={ style }/> }
+			/>
+		</Toolbar>
+	);
+};
+
+const renderContent = ( { backgroundColor, textColor, setBackgroundColor, setTextColor } ) => ( ( { isOpen, onToggle, onClose } ) => {
+	return (
+		<>
+			<div className="color-palette-controller-container">
+				<ColorPaletteControl
+					value={ backgroundColor.color }
+					onChange={ setBackgroundColor }
+					label={ __( 'Background Color' ) }
+				/>
+			</div>
+
+			<div className="color-palette-controller-container">
+				<ColorPaletteControl
+					value={ textColor.color }
+					onChange= { setTextColor }
+					label={ __( 'Text Color' ) }
+				/>
+			</div>
+
+			<ContrastChecker
+				{ ...{
+					textColor: textColor.color,
+					backgroundColor: backgroundColor.color,
+				} }
+				isLargeText={ false }
+			/>
+		</>
+	)
+} );
+
+export default ( { style, className, ...colorControlProps } ) =>
+	<Dropdown
```

Comment:
Have you considered exposing the `Dropdown` as an additionally exposed component? For example, the `URLInput` component is exposed as a standalone component via `index.js`. However an [additional toggle `button` component](https://github.com/WordPress/gutenberg/blob/77c676cf4c1fddc7f81309818840bdf4db0867ba/packages/block-editor/src/components/url-input/button.js#L16) is also exported from a separate file which itself imports `URLInput`.

I wonder if coupling the color picker to a `Dropdown` might be limiting further down the road? Exposing it as an optional extra and allowing the main component to be rendered on its own might be more flexible.

Reply:
We have a couple of options to consider in terms of how to expose these functionality in independent components.
IMHO, we could expose:

1) `<ColorsSelector />` which could be simply what we show into the <Dropdown /> (green:)

2) `<ColorsSelectorToolbar />`, the toolbar included in the block-editor toolbar (red)

<img width="394" alt="Screen Shot 2019-10-16 at 9 24 29 AM" src="https://user-images.githubusercontent.com/77539/66919148-264c3580-eff7-11e9-9c72-51db90594f1e.png">

But I thunk it shuold be done in an other iteration.